presto-spi/src/main/java/com/facebook/presto/spi/plan/MergeJoinNode.java

```diff
@@ -158,4 +159,14 @@ public <R, C> R accept(PlanVisitor<R, C> visitor, C context)
     {
         return visitor.visitMergeJoin(this, context);
     }
+
+    /**
+     *
+     * @param joinNode
+     * @return returns true if merge join is supported for the given join node
+     */
+    public static boolean isMergeJoinEligible(JoinNode joinNode)
+    {
+        return joinNode.getType() == INNER && !joinNode.isCrossJoin();
```

Comment:
I think we can support left join? At least velox support that. Any restriction here? With pre-materialized table on the sequence storage side, we might just need inner join.